curated_transformers/models/albert/_hf.py

```diff
@@ -1,25 +1,46 @@
-import re
-from types import MappingProxyType
-from typing import Any, Callable, Dict, Mapping, Tuple, Union
-
-from torch import Tensor
+from typing import Any, Callable, Dict, List, Tuple, Union
 
 from ...layers.activations import Activation
-from ..hf_hub import _process_hf_keys
+from ...util.string import StringTransform, StrLStrip, StrRepl, StrSub, StrSubInv
+from ..hf_hub.conversion import process_hf_keys
 from .config import ALBERTConfig
 
-HF_KEY_TO_CURATED_KEY = MappingProxyType(
-    {
-        "embeddings.word_embeddings.weight": "embeddings.piece_embeddings.weight",
-        "embeddings.token_type_embeddings.weight": "embeddings.type_embeddings.weight",
-        "embeddings.position_embeddings.weight": "embeddings.position_embeddings.weight",
-        "embeddings.LayerNorm.weight": "embeddings.embed_output_layer_norm.weight",
-        "embeddings.LayerNorm.bias": "embeddings.embed_output_layer_norm.bias",
-        # Embedding projection
-        "encoder.embedding_hidden_mapping_in.weight": "embeddings.projection.weight",
-        "encoder.embedding_hidden_mapping_in.bias": "embeddings.projection.bias",
-    }
-)
+# Order-dependent.
+HF_PARAM_KEY_TRANSFORMS: List[StringTransform] = [
+    # Prefixes.
+    StrLStrip("albert.", reversible=False),
```

Comment:
I wonder if this is a case where a nested class with factory methods works better, so we could have something like:

```
    StringTransform.remove_prefix("albert.", reversible=False),
```

Or to make it clear that a structure is built:

```
    StringTransform.RemovePrefix("albert.", reversible=False),
```

Reply:
Good idea! I like the factory method, given that we already have a base class. Will rewrite the implementation.

Reply:
Added `StringTransformations` to expose the factory methods.